regulation-worker/cmd/main.go

```diff
@@ -51,8 +56,59 @@ func Run(ctx context.Context) error {
 	misc.Init()
 	diagnostics.Init()
 	backendconfig.Init()
-
 	stats.Default.Start(ctx)
+	defer stats.Default.Stop()
+
+	// START - OpenTelemetry setup
```

Comment:
**Note:** We are using rudder-server's `stats` in [data-regulation-service](https://github.com/rudderlabs/data-regulation-service/blob/master/cmd/service/main.go#L36) too. This would mean that in the future, once the dependency version is bumped we will need to adapt the agent stack there as well.

cc @BonapartePC @saurav-malani